crates/eframe/src/native/wgpu_integration.rs

```diff
@@ -488,20 +500,31 @@ impl<'app> WinitApp for WgpuWinitApp<'app> {
 }
 
 impl<'app> WgpuWinitRunning<'app> {
-    fn save_and_destroy(&mut self) {
-        profiling::function_scope!();
-
-        let mut shared = self.shared.borrow_mut();
+    /// Saves the application state. Set `force_save_without_window` to `true` if the state should be saved even if no window is associated to the application (this happens on Android when the `suspended` event is received).
+    fn save(&mut self, force_save_without_window: bool) {
+        let shared = self.shared.borrow();
+        // This is done because of the "save on suspend" logic on Android. Once the application is suspended, there is no window associated to it.
         if let Some(Viewport { window, .. }) = shared.viewports.get(&ViewportId::ROOT) {
+            log::debug!("Saving application state with a window");
             self.integration.save(self.app.as_mut(), window.as_deref());
+        } else if force_save_without_window {
+            log::debug!("Saving application state without a window");
+            self.integration.save(self.app.as_mut(), None);
```

Comment:
Is `force_save_without_window` really necessary? To me it makes more sense that we always call `self.integration.save` whenever `WgpuWinitRunning::save` is called, with or without a window